lib/src/data/datasource/remote/movie_api_provider.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart';
+
+import '../../../core/util/data_state.dart';
+import '../../../core/util/ui_text.dart';
+
+class MovieApiProvider {
+  Client client;
+
+  MovieApiProvider({
+    required this.client,
+  });
+
+  Future<DataState<dynamic>> fetchMovieList(String category) async {
```

Comment:
I've left the fetchMovieGenre as it is because I am returning  a DataState<String> when datasuccess and it can't be assigned to DataState<Map<String, dynamic>>